lucille-core/src/main/java/com/kmwllc/lucille/connector/FileConnector.java

```diff
@@ -0,0 +1,195 @@
+  public static final String CREATED = "file_creation_date";
+  public static final String SIZE = "file_size_bytes";
+  public static final String CONTENT = "file_content";
+
+  // cloudOption Keys
+  public static final String GET_FILE_CONTENT = "getFileContent";
+  public static final String AZURE_CONNECTION_STRING = "connectionString";
+  public static final String AZURE_ACCOUNT_NAME = "accountName";
+  public static final String AZURE_ACCOUNT_KEY = "accountKey";
+  public static final String S3_REGION = "region";
+  public static final String S3_ACCESS_KEY_ID = "accessKeyId";
+  public static final String S3_SECRET_ACCESS_KEY = "secretAccessKey";
+  public static final String GOOGLE_SERVICE_KEY = "pathToServiceKey";
+
+  private static final Logger log = LoggerFactory.getLogger(FileConnector.class);
+
+  private final String pathToStorage;
+  private final Map<String, Object> cloudOptions;
+  private final List<Pattern> includes;
+  private final List<Pattern> excludes;
+  private CloudStorageClient cloudStorageClient;
+  private final URI storageURI;
+  private final boolean getFileContent;
+
+  public FileConnector(Config config) throws ConnectorException {
+    super(config);
+    this.pathToStorage = config.getString("pathToStorage");
+    // compile include and exclude regex paths or set an empty list if none were provided (allow all files)
+    List<String> includeRegex = config.hasPath("includes") ?
+        config.getStringList("includes") : Collections.emptyList();
+    this.includes = includeRegex.stream().map(Pattern::compile).collect(Collectors.toList());
+    List<String> excludeRegex = config.hasPath("excludes") ?
+        config.getStringList("excludes") : Collections.emptyList();
+    this.excludes = excludeRegex.stream().map(Pattern::compile).collect(Collectors.toList());
+    this.getFileContent = config.hasPath("getFileContent") ? config.getBoolean("getFileContent") : true;
+    this.cloudOptions = config.hasPath("cloudOptions") ? config.getConfig("cloudOptions").root().unwrapped() : Map.of();
+    try {
+      this.storageURI = new URI(pathToStorage);
+      log.info("using path {} with scheme {}", pathToStorage, storageURI.getScheme());
+    } catch (URISyntaxException e) {
+      throw new ConnectorException("Invalid path to storage: " + pathToStorage, e);
+    }
+  }
+
+  @Override
+  public void execute(Publisher publisher) throws ConnectorException {
+    if (storageURI.getScheme() != null) {
+      validateCloudOptions(storageURI, cloudOptions);
+      cloudOptions.put(GET_FILE_CONTENT, getFileContent);
+      cloudStorageClient = CloudStorageClient.getClient(storageURI, publisher, getDocIdPrefix(), excludes, includes, cloudOptions);
+      cloudStorageClient.init();
+      cloudStorageClient.publishFiles();
+      try {
+        cloudStorageClient.shutdown();
+      } catch (Exception e) {
+        throw new ConnectorException("Failed to shutdown client.", e);
+      }
```

Comment:
could return here and remove the else block on l105 to reduce nesting

Reply:
updated in [b77e41f](https://github.com/kmwtechnology/lucille/pull/205/commits/b77e41f1e197220437a1fe6654b8194e6be92412)